src/main/kotlin/com/chattriggers/ctjs/minecraft/wrappers/objects/inventory/nbt/NBTTagList.kt

```diff
@@ -7,22 +7,37 @@ class NBTTagList(override val rawNBT: MCNBTTagList) : NBTBase(rawNBT) {
     val tagCount: Int
         get() = rawNBT.tagCount()
 
-    fun appendTag(nbt: NBTBase) = apply {
-        rawNBT.appendTag(nbt.rawNBT)
-    }
+    fun appendTag(nbt: NBTBase) = appendTag(nbt.rawNBT)
 
     fun appendTag(nbt: MCNBTBase) = apply {
         rawNBT.appendTag(nbt)
     }
 
-    operator fun set(id: Int, nbt: NBTBase) {
-        rawNBT.set(id, nbt.rawNBT)
-    }
+    operator fun set(id: Int, nbt: NBTBase) = set(id, nbt.rawNBT)
 
-    operator fun set(id: Int, nbt: MCNBTBase) {
+    operator fun set(id: Int, nbt: MCNBTBase) = apply {
         rawNBT.set(id, nbt)
     }
 
+    fun insertTag(index: Int, nbt: NBTBase) = insertTag(index, nbt.rawNBT)
+
+    fun insertTag(index: Int, nbt: MCNBTBase) = apply {
+        if (index < 0 || index >= tagCount) {
+            throw IndexOutOfBoundsException("Index $index must be between 0 and $tagCount")
+        }
+
+        val list = mutableListOf<MCNBTBase>()
```

Comment:
This feels a bit inefficient, let's just expose `NBTTagList::tagList` using access transformers and call `insert` on that.